sql/parse/parse.go

```diff
@@ -492,25 +489,79 @@ func convertRenameTable(ctx *sql.Context, ddl *sqlparser.DDL) (sql.Node, error)
 }
 
 func convertAlterTable(ctx *sql.Context, ddl *sqlparser.DDL) (sql.Node, error) {
-	switch ddl.ColumnAction {
-	case sqlparser.AddStr:
-		sch, err := tableSpecToSchema(ctx, ddl.TableSpec)
-		if err != nil {
-			return nil, err
+	if ddl.IndexSpec != nil {
+		table := plan.NewUnresolvedTable(ddl.Table.Name.String(), ddl.Table.Qualifier.String())
+		switch ddl.IndexSpec.Action {
+		case sqlparser.CreateStr:
+			var using sql.IndexUsing
+			switch ddl.IndexSpec.Using.Lowered() {
+			case "btree":
+				using = sql.IndexUsing_BTree
+			case "hash":
+				using = sql.IndexUsing_Hash
+			default:
+				using = sql.IndexUsing_Default
+			}
+
+			var constraint sql.IndexConstraint
+			switch ddl.IndexSpec.Type {
+			case sqlparser.UniqueStr:
+				constraint = sql.IndexConstraint_Unique
+			case sqlparser.FulltextStr:
+				constraint = sql.IndexConstraint_Fulltext
+			case sqlparser.SpatialStr:
+				constraint = sql.IndexConstraint_Spatial
+			default:
+				constraint = sql.IndexConstraint_None
+			}
+
+			columns := make([]sql.IndexColumn, len(ddl.IndexSpec.Columns))
+			for i, col := range ddl.IndexSpec.Columns {
+				if col.Length != nil {
```

Comment:
Given that length is unsupported, seems like you should remove it from sql.IndexColumn

Reply:
Honestly that's more of a dolt error. If anything I should probably move the error into dolt, but `go-mysql-server` should pass around key lengths IMO.